packages/contractkit/src/kit.ts

```diff
@@ -125,27 +142,58 @@ export class ContractKit {
     return this.web3.eth.isSyncing()
   }
 
-  sendTransaction(tx: Tx): TransactionResult {
-    const promiEvent = this.web3.eth.sendTransaction({
-      from: this._defaultOptions.from,
-      // TODO this won't work for locally signed TX
-      gasPrice: '0',
-      // @ts-ignore
-      gasCurrency: this._defaultOptions.gasCurrency,
-      // TODO needed for locally signed tx, ignored by now (celo-blockchain with set it)
-      // gasFeeRecipient: this.defaultOptions.gasFeeRecipient,
-      ...tx,
-    })
-    return toTxResult(promiEvent)
+  async sendTransaction(tx: Tx): Promise<TransactionResult> {
+    const finalTx: Tx = this.fillTxDefaults(tx)
+
+    let gas = finalTx.gas
+    if (gas == null) {
+      const inflactionFactor = this.config.gasInflationFactor || 1
+      gas = Math.round((await this.web3.eth.estimateGas({ ...tx })) * inflactionFactor)
+      debug('estimatedGas: %s', gas)
+    }
+
+    return toTxResult(
+      this.web3.eth.sendTransaction({
+        ...finalTx,
+        gas,
+      })
+    )
   }
 
-  sendTransactionObject(
+  async sendTransactionObject(
     txObj: TransactionObject<any>,
-    options?: TxOptions
+    tx?: Omit<Tx, 'data'>
   ): Promise<TransactionResult> {
-    return sendTransaction(txObj, {
-      ...this._defaultOptions,
-      ...options,
-    })
+    const finalTx: Omit<Tx, 'data'> = this.fillTxDefaults(tx)
+
+    let gas = finalTx.gas
+    if (gas == null) {
+      const inflactionFactor = this.config.gasInflationFactor || 1
```

Comment:
>       const inflactionFactor = this.config.gasInflationFactor || 1
Remove `|| 1` since it will never be used. gasInflationFactor is initialized to `1.3`. If you want more robustness, add a check to throw in the case `this.config.gasInflationFactor` is null or undefined.